xwiki-platform-core/xwiki-platform-livedata/xwiki-platform-livedata-webjar/src/main/webjar/Logic.js

```diff
@@ -59,520 +67,763 @@ define(["jquery", "polyfills"], function ($) {
+          if (!condition(e.detail)) { return; }
+        }
+        // call callback
+        callback(e.detail);
+      });
+    },
+
+
+
+
+    /**
+     * ---------------------------------------------------------------
+     * UTILS
+     */
+
+
+    isValidPropertyId: function (propertyId) {
+      return this.data.query.properties.indexOf(propertyId) !== -1;
+    },
 
 
 
 
 
-  /**
-   * Return the property descriptor corresponding to a property id
-   * @param {String} propertyId
-   * @returns {Object}
-   */
-  Logic.prototype.getPropertyDescriptor = function (propertyId) {
-    return this.data.meta.propertyDescriptors.find(function (descriptor) {
-      return descriptor.id === propertyId;
-    });
-  };
+    /**
+     * ---------------------------------------------------------------
+     * DESCRIPTORS
+     */
+
+
+    /**
+     * Return the property descriptor corresponding to a property id
+     * @param {String} propertyId
+     * @returns {Object}
+     */
+    getPropertyDescriptor: function (propertyId) {
+      return this.data.meta.propertyDescriptors.find(function (propertyDescriptor) {
+        return propertyDescriptor.id === propertyId;
+      });
+    },
+
+
+    /**
+     * Return the property type descriptor corresponding to a property id
+     * @param {String} propertyId
+     * @returns {Object}
+     */
+    getPropertyTypeDescriptor: function (propertyId) {
+      var propertyDescriptor = this.getPropertyDescriptor(propertyId);
+      return this.data.meta.propertyTypes.find(function (typeDescriptor) {
+        return typeDescriptor.id === propertyDescriptor.type;
+      });
+    },
+
+
+    /**
+     * Return the property type descriptor corresponding to a property id
+     * @param {String} propertyId
+     * @returns {Object}
+     */
+    getLayoutDescriptor: function (layoutId) {
+      return this.data.meta.layoutDescriptors.find(function (layoutDescriptor) {
+        return layoutDescriptor.id === layoutId;
+      });
+    },
+
+
+    /**
+     * Get the displayer descriptor associated to a property id
+     * @param {String} propertyId
+     */
+    getDisplayerDescriptor: function (propertyId) {
+      if (!this.isValidPropertyId(propertyId)) { return; }
+      // property descriptor config
+      var propertyDescriptor = this.getPropertyDescriptor(propertyId);
+      var propertyDescriptorDisplayer = propertyDescriptor.displayer || {};
+      // property type descriptor config
+      var typeDescriptor = this.getPropertyTypeDescriptor(propertyId);
+      var typeDescriptorDisplayer = typeDescriptor.displayer || {};
+      // default displayer config
+      var displayerId = propertyDescriptorDisplayer.id || typeDescriptorDisplayer.id;
+      var displayer = this.data.meta.displayers.find(function (displayer) {
+        return displayer.id === displayerId;
+      });
+
+      return $.extend({}, displayer, typeDescriptorDisplayer, propertyDescriptorDisplayer);
```

Comment:
Do you mean
`return $.extend(displayer, typeDescriptorDisplayer, propertyDescriptorDisplayer);`
or
`return $.extend({displayer}, typeDescriptorDisplayer, propertyDescriptorDisplayer);` ?

One or the other this would break the code, so I think I misunderstood what you said

Reply:
Oh okay you were speaking of the default displayer literally! :)
Yes I see what you mean for the second part. And also if the propertyDescriptorDisplayer.id is different from typeDescriptorDisplayer.id, we shouldn't use the type descriptor in the merge.
I can do something like that, just before I'd like to have your input on [this review](https://github.com/xwiki/xwiki-platform/pull/1334#discussion_r457904422), so that we can work with default displayer in the correct way.